### PR TITLE
chrysalis: init at 0.7.9

### PR DESCRIPTION
**Pull request by herrwiese**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/chrysalis/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    url = "https://github.com/keyboardio/${pname}/releases/download/${pname}-${version}/${pname}-${version}.AppImage";
+    sha256 = "12w4vv7dwfpvxpc8kpfas90y7yy8mb8dj2096z3vw1bli5lrn3zi";
+  };
+
+  multiPkgs = null;
+  extraPkgs = p: (appimageTools.defaultFhsEnvArgs.multiPkgs p) ++ [
+    p.glib
+  ];
+
+  extraInstallCommands = "mv $out/bin/${name} $out/bin/${pname}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/chrysalis/default.nix:20]:
```suggestion
appimageTools.wrapType2 rec {
  pname = "chrysalis";
  version = "0.7.9";

  src = fetchurl {
    url = "https://github.com/keyboardio/${pname}/releases/download/${pname}-${version}/${pname}-${version}.AppImage";
    sha256 = "12w4vv7dwfpvxpc8kpfas90y7yy8mb8dj2096z3vw1bli5lrn3zi";
  };

  multiPkgs = null;
  extraPkgs = p: (appimageTools.defaultFhsEnvArgs.multiPkgs p) ++ [
    p.glib
  ];

  extraInstallCommands = "mv $out/bin/${name}-binary $out/bin/${pname}";
```

Didn't test.

Comment (herrwiese) [reply pkgs/applications/misc/chrysalis/default.nix:20]: This actually doesn't work, as `appimageTools.wrapType2` isn't a (direct) wrapper around `stdenv.mkDerivation` but `buildFHSUserEnv` which then bails out with unknown arguments (`version`).

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103207` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>chrysalis</li>
  </ul>
</details>

File: pkgs/applications/misc/chrysalis/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+in appimageTools.wrapType2 rec {
+  name = "${pname}-${version}-binary";
+
+  src = fetchurl {
+    url = "https://github.com/keyboardio/${pname}/releases/download/${pname}-${version}/${pname}-${version}.AppImage";
+    sha256 = "12w4vv7dwfpvxpc8kpfas90y7yy8mb8dj2096z3vw1bli5lrn3zi";
+  };
+
+  multiPkgs = null;
+  extraPkgs = p: (appimageTools.defaultFhsEnvArgs.multiPkgs p) ++ [
```
Comment (spacefrogg) [inline pkgs/applications/misc/chrysalis/default.nix:15]: Why do you transfer the packages from `multiPkgs` to `extraPkgs`?

Comment (herrwiese) [reply pkgs/applications/misc/chrysalis/default.nix:15]: Setting `multiPkgs = null` for `appimageTools.wrapType2` disables multilib packages (this is a pure 64-bit package).  `appimageTools.defaultFhsEnvArgs.multiPkgs` holds the list of packages required for an AppImage, which would normally be pulled in by `multiPkgs`, though, and now need to be explicitly pulled in as non-multilib packages.

This is a quite common construct in `nixpkgs` for AppImages (I admit to not having thought this up myself ;)

**Comment by spacefrogg** [comment]
LGTM

**Pull request merged**